classes/proxy_lock_factory.php

```diff
@@ -226,19 +226,30 @@ private function log_lock($resourcekey) {
 
         $record = $DB->get_record_select('tool_lockstats_locks', $select, $params);
 
+        if (strpos($resourcekey, 'adhoc') !== false) {
+            $adhocid = explode('_', $resourcekey);
```

Comment:
This is a little loose and could pick up erroneous stuff, you should be able to reuse some existing helpers:

classes/table/locks.php:139:        preg_match(" /^adhoc_(\d+)$/", $task, $results);
classes/table/locks.php-140-
classes/table/locks.php-141-        if (count($results) > 0) {
classes/table/locks.php-142-            return (int) $results[1];
classes/table/locks.php-143-        } else {
classes/table/locks.php-144-            return null;
classes/table/locks.php-145-        }